flink-python/pyflink/datastream/connectors.py

```diff
@@ -0,0 +1,692 @@
+            .org.apache.flink.streaming.connectors.kafka.FlinkKafkaProducer
+        j_flink_kafka_producer = JFlinkKafkaProducer(
+            topic, serialization_schema._j_serialization_schema, j_properties)
+        super(FlinkKafkaProducer, self).__init__(sink_func=j_flink_kafka_producer)
+
+    def set_write_timestamp_to_kafka(self, write_timestamp_to_kafka: bool):
+        """
+        If set to true, Flink will write the (event time) timestamp attached to each record into
+        Kafka. Timestamps must be positive for Kafka to accept them.
+
+        :param write_timestamp_to_kafka: Flag indicating if Flink's internal timestamps are written
+                                         to Kafka.
+        """
+        self._j_function.setWriteTimestampToKafka(write_timestamp_to_kafka)
+
+    def ignore_failures_after_transaction_timeout(self):
+        """
+        Disables the propagation of exceptions thrown when committing presumable timed out Kafka
+        transactions during recovery of the job. If a Kafka transaction is timed out, a commit will
+        never be successful. Hence, use this feature to avoid recovery loops of the job. Exceptions
+        will still be logged to inform the user that data loss might have occurred.
+
+        Note that we use current time millis to track the age of a transaction. Moreover, only
+        exceptions thrown during the recovery are caught, i.e., the producer will attempt at least
+        one commit of the transaction before giving up.
+        """
+        self._j_function = self._j_function.ignoreFailuresAfterTransactionTimeout()
+        return self
+
+    def set_log_failures_only(self, log_failures_only: bool):
+        """
+        Defines whether the producer should fail on errors, or only log them. If this is set to
+        true, then exceptions will be only logged, if set to false, exceptions will be eventually
+        thrown and cause the streaming program to fail (and enter recovery).
+
+        :param log_failures_only: The flag to indicate logging-only on exceptions.
+        """
+        self._j_function.setLogFailuresOnly(log_failures_only)
+
+    def set_flush_on_checkpoint(self, flush_on_checkpoint: bool):
+        """
+        If set to true, the Flink producer will wait for all outstanding messages in the Kafka
+        buffers to be acknowledged by the Kafka producer on a checkpoint.
+
+        This way, the producer can guarantee that messages in the Kafka buffers are part of the
+        checkpoint.
+
+        :param flush_on_checkpoint: Flag indicating the flush mode (true = flush on checkpoint)
+        """
+        self._j_function.setFlushOnCheckpoint(flush_on_checkpoint)
+
+
+class JdbcSink(SinkFunction):
+
+    def __init__(self, j_jdbc_sink):
+        super(JdbcSink, self).__init__(sink_func=j_jdbc_sink)
+
+    @staticmethod
+    def sink(sql: str, type_info: RowTypeInfo, jdbc_connection_options: 'JdbcConnectionOptions',
```

Comment:
We can make type_info optional too since we can directly get this type from the input datastream. e.g.,
```
    def sink(sql: str, jdbc_connection_options: 'JdbcConnectionOptions', type_info: RowTypeInfo = None,
             jdbc_execution_options: 'JdbcExecutionOptions' = None):
```